Test/Invoke-WhiskeyProGetUniversalPackageTask.Tests.ps1

```diff
@@ -1100,3 +1157,26 @@ Describe 'Invoke-WhiskeyProGetUniversalPackageTask.when packaging a directory' {
     ThenPackageShouldInclude 'dir1\subdir\file.txt'
     ThenPackageShouldNotInclude ('dir1\{0}' -f $defaultPackageName)
 }
+
+Describe 'Invoke-WhiskeyProGetUniversalPackageTask.when compressionLevel of 9 is included' {
+    GivenARepositoryWithFiles 'one.ps1'
+    WhenPackaging -Paths '*.ps1' -WithWhitelist "*.ps1" -CompressionLevel 9
+    ThenPackageShouldbeBeCompressed 'one.ps1' -ExpectedPackageSize 798
+}
+
+Describe 'Invoke-WhiskeyProGetUniversalPackageTask.when compressionLevel is not included' {
+    GivenARepositoryWithFiles 'one.ps1'
+    WhenPackaging -Paths '*.ps1' -WithWhitelist "*.ps1"
+    ThenPackageShouldbeBeCompressed 'one.ps1' -ExpectedPackageSize 809
+}
+
+Describe 'Invoke-WhiskeyProGetUniversalPackageTask.when a bad compressionLevel is included' {
+    GivenARepositoryWithFiles 'one.ps1'
+    WhenPackagingWithBadCompressionLevelShouldThrowAnError -Paths '*.ps1' -WithWhitelist "*.ps1" -CompressionLevel "this is no good"

```

Comment:
Let's do this instead:

```
WhenPackaging -Path '*.ps1' -WithWhitelist '*.ps1' -CompressionLevel 'this is no good'
ThenTaskFails 'not\ a\ valid\ compression\ level'
```

Extract the logic for the `ThenTaskFails` function from the `WhenPackaging` function. Update all the other tests that assumed this would get called to call this assertion explicitly. You'll probably need to create a counter `ThenTaskSucceeds` function (logic extracted from `WhenPackaging`) that all the other tests should use.